design_drafts/async_controller_impl.md

```diff
@@ -0,0 +1,40 @@
+## Implementation of Asynchronous Controllers
+
+
+#### Short description
+
+Asynchronous controllers are an upcoming addition to the ros2_control framework. As the name suggests, they make it possible to run the update calls of these controllers on a separate thread. Asynchronous components are omitted here for the sake of clarity, but they follow a very similar structure.
+Since the intention was to support real-time systems, the standard library's std::atomic interface was used, and due to the inherent difficulty of writing and understanding concurrent lock-free programs, a brief glance at the code might not be enough to figure out if something doesn't work as intended.
+
+For that reason, the purpose of this design draft is to explain how the synchronization works between the Controller Manager's thread and an asynchronous controllers, now to make it easier for others to review the PR of the feature, and later to let the users know if the current implementation is suitable for their use case.
+
+#### Implementation
+
+The controllers running asynchronously are entirely independent of each other, which means that no data sharing occurs between them. Therefore synchronization happens between the Controller Manager's thread and an async controller - so in the following drawings only these two threads will be visualized.
```

Comment:
Here I wanted to say that since each asynchronous controller has its own instance of atomic flags to signal if their update has finished, even if you have 8 async controllers, it's enough to visualize only the controller manager's thread and only a single controller thread to show how synchronization is implemented. Because the controllers don't share data with each other, no synchronization happens between the controllers at all. The CM thread, in turn, communicates with all controller threads, but only with one at a time. In the code it's just a for loop which checks each controller thread's flags in order.

I'm not sure if it's more clear now, but in the evening I'll think about how to reword this paragraph.